packages/patternfly-4/react-core/src/components/Nav/NavList.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import NavGroup from './NavGroup';
+
+const propTypes = {
+  /** HTML ID of the list */
+  id: PropTypes.string,
+  /** Aria label that can be used in conjunction with expandable lists */
+  'aria-label': PropTypes.string,
+  /** Should only contain NavListItem nodes */
+  children(props, propName, componentName) {
+    const prop = props[propName];
+
+    let error = null;
+    React.Children.forEach(prop, child => {
+      if (child.type.componentType !== 'NavListItem') {
+        error = new Error(`${componentName} children should be of type NavListItem but saw ${child.type}.`);
+      }
+    });
+    return error;
+  },
+  /** Additional classes added to the list */
+  className: PropTypes.string,
+  /** Additional classes added to the inner expandable list */
+  innerClassName: PropTypes.string,
+  /** Title shown for sections or expandable lists */
+  title: PropTypes.string,
+  /** True to make the list expandable */
+  isExpandable: PropTypes.bool,
+  /** Flag indicating whether the list is expanded */
+  isExpanded: PropTypes.bool,
+  /** Flag indicating whether the list is active */
+  isActive: PropTypes.bool
+};
+
+const defaultProps = {
+  id: '',
+  'aria-label': '',
+  children: null,
+  className: '',
+  innerClassName: '',
+  title: '',
+  isExpandable: false,
+  isExpanded: false,
+  isActive: false
+};
+
+const renderChildren = (children, groupId, onSelect) =>
```

Comment:
If you're gonna use react context you don't have to copy the children once again and just call `onSelect` from children right away.

Reply:
Not sure how to add index to the children without cloning? I did add context for the onSelect tho.